latch_cli/snakemake/serialize_utils.py

```diff
@@ -217,9 +217,7 @@ def get_serializable_workflow(
 
 
 def update_mapping(cur: Path, stem: Path, remote: str, mapping: Dict[str, str]):
-    if cur.is_file():
-        mapping[str(stem)] = remote
-        return
-
-    for p in cur.iterdir():
-        update_mapping(p, stem / p.name, urljoins(remote, p.name), mapping)
+    if cur.is_dir():
+        for p in cur.iterdir():
+            update_mapping(p, stem / p.name, urljoins(remote, p.name), mapping)
+    mapping[str(stem)] = remote
```

Comment:
Not file != Is Dir, that is the reason for this?

Reply:
Yeah, it only really makes sense to `iterdir` on directories. 

for example, if `cur` points to a device file, we shouldn't iterdir over that file. Symlinks should resolve to the underlying file so it should work as expected for those as well.